cscs-checks/prgenv/hip/build_hip.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class BuildHip(rfm.RegressionTest):
+    '''Download and install HIP around the nvcc compiler.'''
+
+    # HIP build variables
+    hip_path = variable(str, value='hip')
+    hip_full_path = variable(str)
+    hip_platform = variable(str, value='nvcc')
+
+    valid_systems = ['daint:gpu', 'dom:gpu']
+    valid_prog_environs = ['PrgEnv-gnu']
+    sourcesdir = 'https://github.com/ROCm-Developer-Tools/HIP.git'
+    build_system = 'CMake'
+    postbuild_cmds = ['make install']
+    executable = f'{hip_path}/bin/hipcc'
+    executable_opts = ['--version']
+    maintainers = ['JO']
+    tags = {'production'}
+
+    @rfm.run_before('compile')
+    def set_compile_options(self):
+        self.hip_full_path = os.path.abspath(
+            os.path.join(self.stagedir, self.hip_path)
+        )
+        self.build_system.builddir = 'build'
+        self.build_system.config_opts = [
+            f'-DCMAKE_INSTALL_PREFIX={self.hip_full_path}',
```

Comment:
How do you plan to run this test? The `hip_full_path` doesn't point anywhere...

Reply:
What do you mean? `hip_full_path` is defined in line 32, no?

Reply:
Sorry, I missed it!